raft/raft_test.go

```diff
@@ -1278,6 +1278,12 @@ func TestLeaderElectionWithCheckQuorum(t *testing.T) {
 	b.checkQuorum = true
 	c.checkQuorum = true
 
+	// we can not let system choosing the value of randomizedElectionTimeout
+	// otherwise it will introduce some uncertainty into this test case
+	// we need to ensure randomizedElectionTimeout > electionTimeout here
+	a.randomizedElectionTimeout = a.electionTimeout + 1
```

Comment:
guess we can't , and here's the definition of pastElectionTimeout:

```
func (r *raft) pastElectionTimeout() bool {                                                                                    
    return r.electionElapsed >= r.randomizedElectionTimeout                                                                    
} 
```
in your suggestion,  r.electionElapsed = r.randomizedElectionTimeout could be true if randomizedElectionTimeout happen to be electionTimeout(system choose the random part as zero)

then b is gonna campain for itself, it breaks the scenario we set up